tests/Feature/SnmpTraps/CommonTrapTest.php

```diff
@@ -55,138 +63,140 @@ public function testFindByIp()
 UDP: [$ipv4->ipv4_address]:64610->[192.168.5.5]:162
 DISMAN-EVENT-MIB::sysUpTimeInstance 198:2:10:48.91\n";
 
-        Log::shouldReceive('info')->once()->with('Unhandled trap snmptrap', ['device' => $device->hostname, 'oid' => null]);
-        Log::shouldReceive('event')->once()->withArgs(function ($e_message, $e_device, $e_type) use ($device) {
-            return $e_message == '' &&
-                $device->is($e_device) &&
-                $e_type == 'trap';
-        });
+        Log::partialMock()->shouldReceive('info')->once()->with('Unhandled trap snmptrap', ['device' => $device->hostname, 'oid' => null]);
 
         $trap = new Trap($trapText);
         $this->assertFalse(Dispatcher::handle($trap), 'Found handler for trap with no snmpTrapOID');
 
         // check that the device was found
         $this->assertEquals($device->hostname, $trap->getDevice()->hostname);
+
+        // check that eventlog was logged
+        $eventlog = Eventlog::latest('datetime')->first();
+        $this->assertEquals($device->device_id, $eventlog->device_id, 'Trap eventlog device incorrect');
+        $this->assertEquals('', $eventlog->message, 'Trap eventlog message incorrect');
+        $this->assertEquals('trap', $eventlog->type, 'Trap eventlog type incorrect');
```

Comment:
Seems that the eventlog from device creation is hit here in the current failing test  : 
`app/Observers/DeviceObserver.php:        Log::event("Device $device->hostname has been created", $device, 'system', 3);`

https://github.com/librenms/librenms/actions/runs/3371567902/jobs/5593962698

Reply:
Yes, the code simply loads the last eventlog message, but it is not the one expected.  Oddly, this works on my local machine :(